backend/services/services.py

```
@@ -60,26 +99,145 @@
+        raise HTTPException(status_code=404, detail="Post not found or not authorized")
+    
+    event = db.query(Event).filter(Event.post_id == post.id).first()
+    if not event:
+        raise HTTPException(status_code=404, detail=EVENT_404_ERROR)
+    
+    return post, event
+
+def convert_to_utc(event_date: str, event_time: str, user_timezone: str) -> datetime:
+    
+    try:
+        local_datetime = datetime.strptime(f"{event_date} {event_time}", "%Y-%m-%d %H:%M")
```

Ruff (0.8.2)

Naive datetime constructed using `datetime.datetime.strptime()` without %z

(DTZ007)